scripts/generators/feature_requirements.py

```diff
@@ -0,0 +1,215 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+from generators.base_generator import BaseGenerator
+from generators.generator_utils import PlatformGuardHelper
+
+class FeatureRequirementsGenerator(BaseGenerator):
+    def __init__(self):
+        BaseGenerator.__init__(self)
+        
+        # Features of the VkPhysicalDeviceBufferDeviceAddressFeaturesEXT have
+        # the same name as Vulkan 1.2 and
+        # VkPhysicalDeviceBufferDeviceAddressFeaturesKHR features, but are
+        # semantically different.  They are given a suffix to be distinguished.
+        self.identical_but_different_features = {
+            'VkPhysicalDeviceBufferDeviceAddressFeaturesEXT',
+        }
+        
+    def generate(self):
+        out = []
+        out.append(f'''// *** THIS FILE IS GENERATED - DO NOT EDIT ***
+            // See {os.path.basename(__file__)} for modifications
+
+            /***************************************************************************
+            *
+            * Copyright (c) 2023 The Khronos Group Inc.
+            * Copyright (c) 2023 Valve Corporation
+            * Copyright (c) 2023 LunarG, Inc.
+            *
+            * Licensed under the Apache License, Version 2.0 (the "License");
+            * you may not use this file except in compliance with the License.
+            * You may obtain a copy of the License at
+            *
+            *     http://www.apache.org/licenses/LICENSE-2.0
+            *
+            * Unless required by applicable law or agreed to in writing, software
+            * distributed under the License is distributed on an "AS IS" BASIS,
+            * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+            * See the License for the specific language governing permissions and
+            * limitations under the License.
+            ****************************************************************************/\n\n''')
+        out.append('// NOLINTBEGIN\n\n') # Wrap for clang-tidy to ignore
+
+        if self.filename == 'feature_requirements_helper.h':
+            out.append(self.generateHeader())
+        elif self.filename == 'feature_requirements_helper.cpp':
+            out.append(self.generateSource())
+        else:
+            out.append(f'\nFile name {self.filename} has no code to generate\n')
+
+        out.append('\n// NOLINTEND') # Wrap for clang-tidy to ignore
+        self.write("".join(out))
+        
+    def getFeaturesAndOrigins(self):
```

Comment:
I think this is valid to go

```suggestion
    def getFeaturesAndOrigins(self) -> dict:
```

if not, ignore